src/apikey/lib/middlewares/auth.js

```diff
@@ -1,10 +1,15 @@
 const { apikey } = require('config');
 
 /**
- * @param {Object} req - HTTP request
- * @param {Object} res - HTTP response
- * @param {function} next - do the following
- * @returns {Object|function} res or next
+ * Authentication middleware that checks if the content of the x-api-key header
+ * matches the environment variable used as password.
+ *
+ * @param {import('express').Request} req - HTTP request.
+ * @param {import('express').Response} res - HTTP response.
+ * @param {import('express').NextFunction} next - Do the following.
+ *
+ * @return {import('express').Response | import('express').NextFunction}
```

Comment:
This function does not return the `next` callback itself but the result of its call, which is `void`. You could also change the function so that it doesn't return anything, and remove the `@return` statement completely.